File: examples/governance/model_registry_domain/lib/model_registry_domain/ai_model/commands/classify_risk.rb
module ModelRegistryDomain
  class AiModel
    module Commands
      class ClassifyRisk
        include Hecks::Command
        emits "ClassifiedRisk"

        attr_reader :model_id, :risk_level

        def initialize(model_id: nil, risk_level: nil)
          @model_id = model_id
          @risk_level = risk_level
        end

        def call
          existing = repository.find(model_id)
          if existing
            unless existing.status == "draft"
              raise Hecks::Error, "Cannot ClassifyRisk: status must be 'draft', got '#{existing.status}'"
            end
            AiModel.new(
              id: existing.id,
              name: existing.name,
              version: existing.version,
              provider_id: existing.provider_id,
              description: existing.description,
              registered_at: existing.registered_at,
              parent_model_id: existing.parent_model_id,
              derivation_type: existing.derivation_type,
              capabilities: existing.capabilities,
              intended_uses: existing.intended_uses,
              risk_level: risk_level,
              status: "classified"
            )
          else
            raise Hecks::Error, "AiModel not found: #{model_id}"
          end
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/classify_risk.go
package domain

import (
	"time"
	"fmt"
)

type ClassifyRisk struct {
	ModelId string `json:"model_id"`
	RiskLevel string `json:"risk_level"`
}

func (c ClassifyRisk) CommandName() string { return "ClassifyRisk" }

func (c ClassifyRisk) Execute(repo AiModelRepository) (*AiModel, *ClassifiedRisk, error) {
	existing, err := repo.Find(c.ModelId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("AiModel not found: %s", c.ModelId)
	}
	existing.RiskLevel = c.RiskLevel
	if existing.Status != "draft" {
		return nil, nil, fmt.Errorf("cannot ClassifyRisk: AiModel is in %s state", existing.Status)
	}
	existing.Status = "classified"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := ClassifiedRisk{
		AggregateID: existing.ID,
		ModelId: c.ModelId,
		RiskLevel: c.RiskLevel,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}
